src/Dialect/ONNX/Transforms/QDQAroundOpOpt.cpp

```diff
@@ -0,0 +1,132 @@
+//===- QDQOpt.cpp - Remove QDQ operations --------*- C++ -*-===//
+//
+// (c) Copyright 2022 - 2025 Advanced Micro Devices, Inc. All Rights Reserved.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include <cmath>
+#include <mlir/IR/IRMapping.h>
+#include <mlir/IR/Operation.h>
+#include <mlir/IR/PatternMatch.h>
+#include <mlir/Pass/Pass.h>
+#include <src/Dialect/ONNX/ONNXOps.hpp>
+#include <src/Dialect/ONNX/ONNXOps/OpHelper.hpp>
+
+using namespace mlir;
+using namespace onnx_mlir;
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXTransposeOp transposeOp) {
+  return {transposeOp.getData(), transposeOp.getTransposed()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXUnsqueezeOp unsqueezeOp) {
+  return {unsqueezeOp.getData(), unsqueezeOp.getExpanded()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXSqueezeOp squeezeOp) {
+  return {squeezeOp.getData(), squeezeOp.getSqueezed()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXReshapeOp reshapeOp) {
+  return {reshapeOp.getData(), reshapeOp.getReshaped()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXGatherOp gatherOp) {
+  return {gatherOp.getData(), gatherOp.getOutput()};
+}
+// std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+//     ONNXReduceSumOp reduceOp) {
+//   return {reduceOp.getData(), reduceOp.getReduced()};
+// }
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXSliceOp sliceOp) {
+  return {sliceOp.getData(), sliceOp.getOutput()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXResizeOp resizeOp) {
+  return {resizeOp.getX(), resizeOp.getY()};
+}
+std::tuple<Value /*input*/, Value /*output*/> getDataInputOutput(
+    ONNXFlattenOp flattenOp) {
+  return {flattenOp.getInput(), flattenOp.getOutput()};
+}
+namespace {
+template <typename T>
+class RemoveQDQAroundOpPattern : public OpRewritePattern<T> {
+public:
+  using OpRewritePattern<T>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(
+      T op, PatternRewriter &rewriter) const override {
+    Value input, output;
+    std::tie(input, output) = getDataInputOutput(op);
+
+    auto dqOp = input.getDefiningOp<ONNXDequantizeLinearOp>();
+    if (!dqOp) {
+      return failure();
+    }
+    if (output.hasOneUse()) {
```

Comment:
You could save some code indentation by using an early exit
```suggestion
    if (!output.hasOneUse()) {
      return failure();
    }
```